src/client.ts

```diff
@@ -32,9 +32,12 @@ type OnUserUpdateHandler = (userInfo: UserInfo) => void;
 
 type OnForcedVariationsUpdateHandler = () => void;
 
+type NotReadyReason = 'TIMEOUT' | 'NO_CLIENT';
+
 export type OnReadyResult = {
   success: boolean;
-  reason?: string;
+  reason?: NotReadyReason;
+  reasonDetail?: string;
```

Comment:
A Non binding suggestion: Does it make sense to call it `message` instead of `reasonDetail`? This will make the success case look more inuitive where we dont provide a reason but only reasonDetail right now. But its just a personal preference.. Totally non binding.